File: 자프링-GraphQL-실습/src/main/java/com/junyss/graphqltest/api/role/resolver/RoleResolver.java
package com.junyss.graphqltest.api.role.resolver;

import org.springframework.graphql.data.method.annotation.Argument;
import org.springframework.graphql.data.method.annotation.MutationMapping;
import org.springframework.graphql.data.method.annotation.QueryMapping;
import org.springframework.stereotype.Controller;

import com.junyss.graphqltest.api.role.model.dto.request.RoleRequestDto;
import com.junyss.graphqltest.api.role.model.dto.request.RoleUpdateRequestDto;
import com.junyss.graphqltest.api.role.model.dto.response.RoleResponseDto;
import com.junyss.graphqltest.api.role.service.RoleService;
import com.junyss.graphqltest.common.constant.DefaultResponse;

import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
@Controller
public class RoleResolver {

	private final RoleService roleService;

	@MutationMapping
	public DefaultResponse<String> saveForRole (@Argument("input") RoleRequestDto roleRequestDto) {
		return roleService.saveForRole(roleRequestDto);
	}

	@QueryMapping
	public DefaultResponse<RoleResponseDto> getRoleByRoleId (@Argument String roleId) {
		return roleService.getRoleByRoleId(roleId);
	}

	@MutationMapping
	public DefaultResponse<String> updateRole (@Argument("input") RoleUpdateRequestDto roleUpdateRequestDto) {
		return roleService.updateRole(roleUpdateRequestDto);
	}

	@MutationMapping
	public DefaultResponse<String> deleteRole (@Argument String roleId) {
		return roleService.deleteRole(roleId);
	}
}

File: kotlin-graphql-test/src/main/kotlin/com/junyharangstudy/kotlingraphqltest/api/role/resolver/RoleResolver.kt
package com.junyharangstudy.kotlingraphqltest.api.role.resolver

import com.junyharangstudy.kotlingraphqltest.api.common.constant.PageRequestDto
import com.junyharangstudy.kotlingraphqltest.api.role.model.dto.request.RoleCreateRequestDto
import com.junyharangstudy.kotlingraphqltest.api.role.model.dto.request.RoleSearchRequestDto
import com.junyharangstudy.kotlingraphqltest.api.role.model.dto.request.RoleUpdateRequestDto
import com.junyharangstudy.kotlingraphqltest.api.role.model.dto.response.RoleResponseDto
import com.junyharangstudy.kotlingraphqltest.api.role.service.RoleService
import com.junyharangstudy.kotlingraphqltest.api.software.model.dto.response.SoftwareResponseDto
import com.junyharangstudy.kotlingraphqltest.common.constant.DefaultResponse
import org.springframework.graphql.data.method.annotation.Argument
import org.springframework.graphql.data.method.annotation.MutationMapping
import org.springframework.graphql.data.method.annotation.QueryMapping
import org.springframework.stereotype.Controller

@Controller
class RoleResolver (private val roleService: RoleService) {
    @MutationMapping
    fun saveRole(@Argument("input") roleCreateRequestDto: RoleCreateRequestDto): DefaultResponse<String> {
        return roleService.saveRole(roleCreateRequestDto)
    }

    @QueryMapping
    fun getRoleByRoleId (@Argument roleId: String): DefaultResponse<RoleResponseDto> {
        return roleService.getRoleByRoleId(roleId)
    }

    @MutationMapping
    fun updateRole(
        @Argument("roleId") roleId: String,  // softwareId 는 수정할 자료의 고유 ID 입니다. 수정할 자료가 없다면 예외가 발생합니다. 수정할 자료가 여러 개라면 예외가 발생합니다. 수정할 자료가 없다면 예외가 발생합니다. 수정할 자료가 여러 개라면 예외가 발생합니다. 수정할 자료
        @Argument("input") roleUpdateRequestDto: RoleUpdateRequestDto
    ): DefaultResponse<String> {
        return roleService.updateRole(roleId, roleUpdateRequestDto)
    }

    @MutationMapping
    fun deleteRole(@Argument roleId: String): DefaultResponse<String> {
        return roleService.deleteRole(roleId)
    }
}